### PR TITLE
[multiplatform] use docker buildx

### LINKED ISSUE #92: ARM64 Image
I would like to run a development environment with Janusgraph, Solr and Cassandra on my Raspberry Pies. Solr and Cassandra both provide ARM64 compatible images on Docker Hub. 

Sadly Janusgraph only provides AMD images.

It would be nice if Janusgraph had also compatible images. 

Thank you for your feedback.

### PR DESCRIPTION
**Pull request by lionelfleury**
Closes #92

see: https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -27,6 +27,9 @@ jobs:
         version: ["0.5", "0.6"]
     steps:
     - uses: actions/checkout@v2
+    - uses: crazy-max/ghaction-docker-buildx@v1
```
Comment (FlorianHockmann) [inline .github/workflows/deploy.yml:30]: The blog post in your PR description seems to be already a bit outdated and this action has been replaced in the meantime by an official one from Docker. [The readme from this action](https://github.com/crazy-max/ghaction-docker-buildx#moved-to-docker-organization) recommends to use this instead:

```yaml
# https://github.com/docker/setup-qemu-action
- name: Set up QEMU
  uses: docker/setup-qemu-action@v1
# https://github.com/docker/setup-buildx-action
- name: Set up Docker Buildx
  uses: docker/setup-buildx-action@v1
```

and these two actions also already have [newer versions](https://github.com/docker/setup-buildx-action#with-qemu):

```yaml
-
  name: Set up QEMU
  uses: docker/setup-qemu-action@v2
-
  name: Set up Docker Buildx
  id: buildx
  uses: docker/setup-buildx-action@v2
```

**Review by FlorianHockmann — COMMENTED** [review]
Thanks a lot for tackling this, @lionelfleury! I didn't know that it's so easy to build images for multiple architectures 😃 

In addition to my inline comment, could you please add a short note for other contributors that Docker Buildx is used with a link to [the official docs](https://docs.docker.com/buildx/working-with-buildx/)?
Buildx seems to be included and enabled by default already but I think it would be good for contributors to know that this is used in case they run into any errors that require additional config. I had problems with this at first that were probably specific to my setup with Win 10 and WSL 2.

**Comment by lionelfleury** [comment]
done ✅ 
thanks for the feedbacks and versions.

File: BUILDING.md [hunk 2]
```diff
@@ -3,7 +3,8 @@
 ## Building
 
 The `build-images.sh` script will build the docker images for all the Dockerfiles in the versioned
-folder directories.
+folder directories using `buildx` command for multi-plaform.
+See (Docker Buildx)[https://docs.docker.com/buildx/working-with-buildx] for more.
```
Comment (FlorianHockmann) [inline BUILDING.md:7]:
```suggestion
See [Docker Buildx](https://docs.docker.com/buildx/working-with-buildx) for more.
```

Comment (lionelfleury) [reply BUILDING.md]: my bad 😆 thanks for spotting 👍

**Comment by lionelfleury** [comment]
fixed.

**Review by FlorianHockmann — APPROVED** [review]

**Comment by lionelfleury** [comment]
because of `test-images.sh` and calls to `docker tag`, this will need a bit of refinement.
closing for now.

**Comment by analytically** [comment]
@lionelfleury is this still being considered? Not having ARM64 images for M1 is a blocker for many devs with local setups

File: build-images.sh [hunk 3]
```diff
@@ -34,11 +35,11 @@ echo "IMAGE_NAME: ${IMAGE_NAME}"
 for v in "${versions[@]}"; do
   if [ -z "${version}" ] || [ "${version}" == "${v}" ]; then
     # prepare docker tags
-    full_version="$(grep "ARG JANUS_VERSION" ${v}/Dockerfile | head -n 1 | cut -d"=" -f 2)"
+    full_version="$(grep "ARG JANUS_VERSION" "${v}/Dockerfile" | head -n 1 | cut -d"=" -f 2)"
     full_version_with_revision="${full_version}-${REVISION}"
 
     # build and test image
-    docker build -f "${v}/Dockerfile" -t "${IMAGE_NAME}:${full_version}" ${v} --build-arg REVISION="$REVISION" --build-arg CREATED="$CREATED"
+    docker buildx build --platform "${PLATFORMS}" -f "${v}/Dockerfile" -t "${IMAGE_NAME}:${full_version}" "${v}" --build-arg REVISION="$REVISION" --build-arg CREATED="$CREATED" -o type=image
```
Comment (analytically) [inline build-images.sh:42]: docker buildx build supports a --tag flag

screenshot 2022-10-04 at 17 04 31@2x

@lionelfleury

**Pull request closed without merging**